src/cljs/main/broadfcui/common/method/config_io.cljs

```diff
@@ -0,0 +1,139 @@
+
+
+(react/defc IOTables
+  {:start-editing
+   (fn [{:keys [props state locals]}]
+     (swap! locals merge (:values props))
+     (swap! state assoc :editing? true))
+   :cancel-editing
+   (fn [{:keys [state]}]
+     (swap! state dissoc :editing?))
+   :save
+   (fn [{:keys [state locals]}]
+     (swap! state dissoc :editing?)
+     @locals)
+   :render
+   (fn [{:keys [props state this]}]
+     (let [id (gensym "io-table-")]
+       [:div {:id id :style (:style props)}
+        (when (:editing? @state)
+          [:style {} (str ".select2-results__option" ;; TODO - just make this global?
+                          "{font-size: 80%}"
+                          "#" id " .select2-container--default .select2-selection--multiple .select2-selection__choice__remove"
+                          "{display: none}"
+                          "#" id " .select2-container .selection li.select2-selection__choice"
+                          "{background: none; color: black; margin-right: 0; padding: 0}"
+                          "#" id " .select2-selection__rendered"
+                          "{padding-top: 0.1rem}"
+                          "#" id " .select2-selection__rendered > li+li"
+                          "{width: 10px}")])
+        [Collapse {:title "Inputs"
+                   :contents (this :-render-table :inputs)}]
+        [Collapse {:style {:marginTop "1rem"}
+                   :title "Outputs"
+                   :contents (this :-render-table :outputs)}]]))
+   :-render-table
+   (fn [{:keys [props state locals]} io-key]
+     (let [{:keys [inputs-outputs values invalid-values data]} props
+           {:keys [editing?]} @state]
+       [Table {:data (->> (io-key inputs-outputs)
+                          (map (fn [{:keys [name inputType outputType optional] :as item}]
+                                 (let [[task variable] (take-last 2 (string/split name "."))
+                                       k-name (keyword name)
+                                       error-message (get-in invalid-values [io-key k-name])
+                                       value (get-in values [io-key k-name])]
+                                   (merge (dissoc item :inputType :outputType :optional)
+                                          (utils/restructure task variable error-message value)
+                                          {:type (process-type (or inputType outputType))
+                                           :optional? optional})))))
+               :body {:empty-message (str "No " (string/capitalize (name io-key)))
+                      :style (merge
+                              table-style/table-light
+                              {:body-cell {:padding 0}
+                               :header-row {:borderBottom style/standard-line}
+                               :body-row (constantly {:margin "4px 0" :alignItems "baseline"})})
+                      :behavior {:filterable? false :reorderable-columns? false :allow-no-sort? true
```

Comment:
until we order inputs/outputs by the order they exist in the WDL, could we give an initial sort by task, then variable?